internal/proxy/proxy.go

```diff
@@ -195,8 +198,17 @@ func (p *Proxy) handleJenkinsUIRequest(w http.ResponseWriter, r *http.Request) (
 					err = p.processTemplate(w, ns)
 					p.RecordStatistics(pci.NS, time.Now().Unix(), 0) //FIXME - maybe do this at the beginning?
 				} else { //If Jenkins is running, remove the cookie
-					cookie.Expires = time.Unix(0, 0)
-					http.SetCookie(w, cookie)
+					var resp *http.Response
```

Comment:
AFAIK you need to close the response body. Check the godoc or see here https://stackoverflow.com/questions/33238518/what-could-happen-if-i-dont-close-response-body-in-golang.

You should be able to use _defer_ in _loginJenkins_.

That said, do you really need to return the full reponse? Could you not just return status code and error?

Reply:
Good catch, I'll fix it

The problem with just returning the code is that I used the same function here https://github.com/fabric8-services/fabric8-jenkins-proxy/pull/116/files#diff-80a67e0d6b63df2dd084ee5a102f2f62R262 where we need cookies, so we'd have to return statusCode, cookies and err - I can do that if you think it's better